h5netcdf/core.py

```diff
@@ -400,6 +444,10 @@ def _create_dim_scales(self):
                 dims = self._variables[dim].dimensions
                 coord_ids = np.array([dim_order[d] for d in dims], 'int32')
                 h5ds.attrs['_Netcdf4Coordinates'] = coord_ids
+            # Might have to be resized if it already exists.
```

Comment:
How would this situation come up? My current thinking is that either:

- The variable corresponding to the dimension scale already existed. If any resizing was done, the variable also got resized via `resize_dimension()`.
- If the dimension scale didn't exist, it's created here with the `create_dataset`.

Reply:
You are right - a remnant from the old logic - the data sets can be created with the correct size right away.